server/plugin.go

```diff
@@ -151,6 +163,54 @@ func signClaims(secret string, claims *Claims) (string, error) {
 	return string(token.Raw()), nil
 }
 
+func signClaimsJaaS(apiKeyJaaS string, privateKeyJaaS string, claimsJaaS *JaaSClaims) (string, error) {
+	var err error
+	privateKeyJaaSBytes := []byte(privateKeyJaaS)
+	privPem, _ := pem.Decode(privateKeyJaaSBytes)
+	if privPem == nil {
+		mlog.Error("Error decoding specified JaaS private key")
+		return "", errors.New("error decoding specified JaaS private key")
+	}
+
+	privPemBytes := privPem.Bytes
+	var parsedKey interface{}
+	switch privPem.Type {
+	case rsaPrivateKey:
+		parsedKey, err = x509.ParsePKCS1PrivateKey(privPemBytes)
+	case privateKey:
+		parsedKey, err = x509.ParsePKCS8PrivateKey(privPemBytes)
+	default:
+		return "", errors.New("invalid private key")
+	}
+
+	if err != nil {
+		mlog.Error("Error parsing JaaS private key", mlog.Err(err))
+		return "", err
```

Comment:
To ensure the error is always in the log message and response, we can add the message to the error, and then have the caller of this function log the error

```suggestion
		return errors.Wrap(err, "Error parsing JaaS private key")
```